core/dbt/contracts/graph/parsed.py

```diff
@@ -221,8 +224,34 @@ class ParsedNodeDefaults(NodeInfoMixin, ParsedNodeMandatory):
     unrendered_config: Dict[str, Any] = field(default_factory=dict)
     created_at: float = field(default_factory=lambda: time.time())
     config_call_dict: Dict[str, Any] = field(default_factory=dict)
+    constraints_enabled: bool = False
+
+    def constraints_validator(self):
+        materialization_error = {}
+        language_error = {}
+        data_type_errors = {}
+
+        if self.resource_type == "model" and self.config.constraints_enabled is True:
+            if self.config.materialized != "table":
+                materialization_error = {"materialization": self.config.materialized}
+
+            language = str(self.language)
+            if language != "sql":
+                language_error = {"language": language}
+
+            for column, column_info in self.columns.items():
+                if column_info.data_type is None:
+                    data_type_error = {column: {"data_type": column_info.data_type}}
+                    data_type_errors.update(data_type_error)
+
+        if materialization_error or language_error or data_type_errors:
+            raise CompilationException(
+                f"Only the SQL table materialization is supported for constraints. \n`data_type` values must be defined for all columns and NOT be null or blank.\n  Materialization Error: {materialization_error}\n  Language Error: {language_error}\n  Data Type Errors: {data_type_errors}"
```

Comment:
Would we want to restrict the message to the specific type of error? i.e. not showing `materialization_error` if the error is about data type?

Reply:
Yep I was just thinking about this last night! I'll figure out how to make conditional errors display!